include/openthread/link.h

```diff
@@ -1056,6 +1056,16 @@ otError otLinkCslSetChannel(otInstance *aInstance, uint8_t aChannel);
  */
 uint16_t otLinkCslGetPeriod(otInstance *aInstance);
 
+/**
+ * This function gets the CSL period in milliseconds.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ *
+ * @returns The CSL period in units of milliseconds.
+ *
+ */
+uint32_t otLinkCslGetPeriodMs(otInstance *aInstance);
```

Comment:
We already have existing get/set APIs, but in units of 10 symbols. In general, I think it's better to have only one way of doing things. What about providing a convenience macro that converts ms to units of 10 symbols instead?

Reply:
Doing it only one way is certainly preferable. Do you envision something as simple as below, added to link.h?
```
#define SET_CSL_PERIOD_MS(aInstance, aPeriod) setCslPeriod(aInstance, aPeriod * 1000 / OT_US_PER_TEN_SYMBOLS)
#define GET_CSL_PERIOD_MS(aInstance) getCslPeriod(aInstance) * OT_US_PER_TEN_SYMBOLS / 1000
```

However, I dont see how the later may replace Get<Mac::Mac>().GetCslPeriodMs() which is called also in data_poll_sender.cpp and mle.cpp not importing link.h?

Also cli.cpp does a conversion to ms using multiplication with `kUsPerTenSymbols / 1000`, and may instead also call `otLinkCslGetPeriodMs`. Also, when having `otLinkCslGetPeriodMs` and `otLinkCslSetPeriodMs`, then `OT_US_PER_TEN_SYMBOLS` may get removed from link.h, right?

In order to do it one way only - can we change the existing  `otLinkCslGetPeriod` and `otLinkCslSetPeriod` to take millisecond as arguments?

Reply:
I see another related constraint with respect to cslPeriod in todays implementation not yet addressed: The maximum value of cslPeriod is currently limited to ~10.5 seconds while pollPeriod could be set to hours. A larger value for cslPeriod eg. in the range of minutes is desirable for some applications. Could we change mCslPeriod and the get/set CslPeriod API from 16-bit to 32-bit value?

Reply:
@martinzi , apologies for the delay in responding to this.

We just submitted the following PR to help address what you are looking for:
- https://github.com/openthread/openthread/pull/9085

Can you indicate whether the above PR does indeed address your needs?

Reply:
Above PR LGTM. Thank you.

Reply:
I have removed the millisecond APIs from this PR.